### DISCUSSION TITLE
Working with aiomysql

### DISCUSSION THREAD

**Discussion by gigiallan** [planetscale/discussion#115] [FAQs] [answered]
I'm using aiomysql (async MySQL driver in python) to connect Planetscale, are there any examples or suggestions for me. I search in the documentation but cannot find anything related to async python MySQL drivers.

**Comment by johnyerhot** [comment, marked as answer]
Hey @gigiallan  - sure a good example is [right here](https://github.com/aio-libs/aiomysql/blob/0a8af2355562ea961477738d66b3334332447186/examples/example_ssl.py). Note that you'll need to make sure to use TLS. 

A fully working example for `aiomysql` would look like this:

```python
import asyncio
import aiomysql
import ssl


ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
# the root CA path is dependent on your system. 
ctx.load_verify_locations(cafile='/etc/ssl/cert.pem')

async def test_example(loop):
    pool = await aiomysql.create_pool(host='<planetscale-db-host>', port=3306,
                                      user='<planetscale-db-username>', password='<planetscale-db-password>',
                                      db='<planetscale-db-name>', loop=loop, ssl=ctx)
 async with pool.acquire() as conn:
        async with conn.cursor() as cur:
            await cur.execute("SELECT 42;")
            (r,) = await cur.fetchone()
            print(r)
    pool.close()
    await pool.wait_closed()


loop = asyncio.get_event_loop()
loop.run_until_complete(test_example(loop)
```

You should get `42` back.

**Reply by eliac7** [reply]
I couldn't connect with Planetscale when testing the code on a Windows system. Adding the following code allowed me to finally connect:

```
if os.name == 'nt':
    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
```
Could finally connect.
